src/file.py

```diff
@@ -0,0 +1,406 @@
+"""
+file.py
+
+File class to interact with the CS3 API.
+"""
+
+import time
+import logging
+import http
+import requests
+import cs3.storage.provider.v1beta1.resources_pb2 as cs3spr
+import cs3.storage.provider.v1beta1.provider_api_pb2 as cs3sp
+import cs3.gateway.v1beta1.gateway_api_pb2_grpc as cs3gw_grpc
+import cs3.rpc.v1beta1.code_pb2 as cs3code
+import cs3.types.v1beta1.types_pb2 as types
+import common
+
+from configparser import ConfigParser
+from typing import Iterator, Union
+from exceptions.exceptions import AuthenticationException
+from cs3resource import Resource
+from auth import Auth
+
+
+class File:
+    """
+    File class to interact with the CS3 API.
+    """
+
+    def __init__(
+        self, config: ConfigParser, log: logging.Logger, gateway: cs3gw_grpc.GatewayAPIStub, auth: Auth
+    ) -> None:
+        self.log: logging.Logger = log
+        self.config: ConfigParser = config
+        self.auth: Auth = auth
+        self.gateway: cs3gw_grpc.GatewayAPIStub = gateway
+        self.ssl_verify: bool = config.getboolean("cs3", "sslverify", fallback=True)
```

Comment:
hmm I see this almost contradicts what I mentioned in https://github.com/cernbox/cs3-python-client/pull/1#discussion_r1686746082 :)

I still see the configuration (file) as something owned by the application, and the library here should only access the needed parameters and use appropriate defaults. As such it might be an overkill to have a Config class. One (mild) advantage for keeping the reference to the `ConfigParser` instance given as argument is that if the config changes it is re-read on the fly without requiring a restart of the application.

Reply:
I have an idea, if we make the Config class take the ConfigParser as a parameter (as in the updated version) and make each of the attributes properties that are read from the ConfigParser instance, e.g. 

`
@property
def ssl_verify(self): -> bool:
    return config.getboolean(self.category, "sslverify", fallback=True)
`

That way the values won't be set in each class and we still get the advantage you are talking about Giuseppe.

Reply:
I agree in letting the config handling being done by the app (so, if it wants to retrieve it from a file, it needs to implement such a thing).
But I think we should still have a central way to set defaults, instead of putting them everywhere. Can't we just create an internal config that inherits from `ConfigParser`?

Reply:
Ah, your approach seems a solution, @rawe0.. So go ahead with that.